graylog2-server/src/main/java/org/graylog2/inputs/InputServiceImpl.java

```diff
@@ -142,6 +142,9 @@ public Input create(Map<String, Object> fields) {
 
     @Override
     public Input find(String id) throws NotFoundException {
+        if (!ObjectId.isValid(id)) {
+            throw new NotFoundException("Input id <" + id + "> is invalid!");
```

Comment:
`InputService` and its implementation are actually quite technical. The high-level access is channeled through the classes calling the methods of `InputService`, e. g. `InputsResource`.

It would even be better to have a proper JAX-RS and Jackson serializer and deserializer for `ObjectId` which would respond with an appropriate HTTP status code (e. g. 400 Bad Request) if no valid `ObjectId` was provided. This way it wouldn't only be "point repair" in `InputService` but could be used everywhere.